packages/core/src/common/message-rpc/msg-pack-extension-manager.ts

```diff
@@ -0,0 +1,87 @@
+// *****************************************************************************
+// Copyright (C) 2022 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { addExtension } from 'msgpackr';
+import { ResponseError } from './rpc-message-encoder';
+
+/**
+ * Handles the global registration of custom MsgPackR extensions
+ * required for the default RPC communication. MsgPackR extensions
+ * are installed globally on both ends of the communication channel.
+ * (frontend-backend, pluginExt-pluginMain).
+ * Is implemented as singleton as it is  also used in plugin child processes which have no access to inversify.
+ */
+export class MsgPackExtensionManager {
```

Comment:
I agree, the global extension mechanism is not ideal but sufficient for now. Maybe we should raise an issue for that in the msgpackR Repo. Development there seems to pretty active and responsive to community feedback. So we might be able to get a better extension mechanism with future versions.

Here is my reasoning for introducing this class:
- Better error handling. MsgpackR simply overrides previous registered extensions if you add a new one with the same tag. This might make error tracing really difficult (which is already complicated enough in the rpc context). With the dedicated manager we can throw errors for extensions with duplicate or invalid (out-of-range) tags.
- MsgPackR provides no possibility to retrieve already registered extensions.
- Adding and extension would require an direct import from "msgPackR". The dependency is currently not exported via @theia/core/shared so this opens up the possibility for version resolution issues. All we need is basically one function and one type definition, so wrapping and reexporting this in Theia core via `MsgPackExtensionManager` might be cleaner.
- Adopters don't have to interact directly with a third party dependency
- The msgpackR source (especially the typings) is not always well documented. With this approach we can keep documentation for the extension mechanism in repo.
However, I have no strong opinion on that and I'm also fine to use msgPackR directly.